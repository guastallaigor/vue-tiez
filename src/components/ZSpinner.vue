<template>
  <div class="z-spinner" :style="{width: size, height: size}" :class="{ dark }">
    <div class="spinner" :style="{width: size, height: size}">
      <div class="spinner_bar" style="animation-delay: -1.2s; transform: rotate(0deg) translate(146%);"></div>
      <div class="spinner_bar" style="animation-delay: -1.1s; transform: rotate(30deg) translate(146%);"></div>
      <div class="spinner_bar" style="animation-delay: -1s; transform: rotate(60deg) translate(146%);"></div>
      <div class="spinner_bar" style="animation-delay: -0.9s; transform: rotate(90deg) translate(146%);"></div>
      <div class="spinner_bar" style="animation-delay: -0.8s; transform: rotate(120deg) translate(146%);"></div>
      <div class="spinner_bar" style="animation-delay: -0.7s; transform: rotate(150deg) translate(146%);"></div>
      <div class="spinner_bar" style="animation-delay: -0.6s; transform: rotate(180deg) translate(146%);"></div>
      <div class="spinner_bar" style="animation-delay: -0.5s; transform: rotate(210deg) translate(146%);"></div>
      <div class="spinner_bar" style="animation-delay: -0.4s; transform: rotate(240deg) translate(146%);"></div>
      <div class="spinner_bar" style="animation-delay: -0.3s; transform: rotate(270deg) translate(146%);"></div>
      <div class="spinner_bar" style="animation-delay: -0.2s; transform: rotate(300deg) translate(146%);"></div>
      <div class="spinner_bar" style="animation-delay: -0.1s; transform: rotate(330deg) translate(146%);"></div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'z-spinner',
  props: {
    size: {
      type: String,
      default: '20px'
    },
    dark: {
      type: Boolean,
      default: false
    }
  },
  data: () => ({})
}
</script>
