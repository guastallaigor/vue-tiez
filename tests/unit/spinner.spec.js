import { shallowMount } from '@vue/test-utils'
import ZSpinner from '../../src/components/ZSpinner'

describe('When I create the ZSpinner component', () => {
  const createComponent = (propsData = {}, slot) => {
    return shallowMount(ZSpinner, {
      propsData,
      slots: {
        default: slot || 'ZSpinner'
      }
    })
  }

  it('should be a Vue instance', () => {
    const wrapper = createComponent()
    expect(wrapper.find(ZSpinner).isVueInstance()).toBe(true)
  })

  it('should have a size of 20px', () => {
    const wrapper = createComponent()
    const spinner = wrapper.find('.z-spinner')
    expect(spinner.exists()).toBe(true)
    expect(spinner.classes().length).toBe(1)
    expect(spinner.classes()).toEqual(['z-spinner'])
    expect(spinner.attributes().style).toBe('width: 20px; height: 20px;')
    const spinnerChild = wrapper.find('.z-spinner > .spinner')
    expect(spinnerChild.exists()).toBe(true)
    expect(spinnerChild.attributes().style).toBe('width: 20px; height: 20px;')
  })

  it('should have a size of 666px', () => {
    const wrapper = createComponent({ size: '666px' })
    const spinner = wrapper.find('.z-spinner')
    expect(spinner.exists()).toBe(true)
    expect(spinner.classes().length).toBe(1)
    expect(spinner.classes()).toEqual(['z-spinner'])
    expect(spinner.attributes().style).toBe('width: 666px; height: 666px;')
    const spinnerChild = wrapper.find('.z-spinner > .spinner')
    expect(spinnerChild.exists()).toBe(true)
    expect(spinnerChild.attributes().style).toBe('width: 666px; height: 666px;')
  })

  it('should have a dark class', () => {
    const wrapper = createComponent({ dark: true })
    const spinner = wrapper.find('.z-spinner')
    expect(spinner.exists()).toBe(true)
    expect(spinner.classes().length).toBe(2)
    expect(spinner.classes()).toEqual(['z-spinner', 'dark'])
  })

  it('should match snapshot', () => {
    const wrapper = createComponent({
      size: '20px',
      dark: false
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
