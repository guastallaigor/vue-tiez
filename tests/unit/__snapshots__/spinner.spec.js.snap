// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I create the ZSpinner component should match snapshot 1`] = `
<div class="z-spinner" style="width: 20px; height: 20px;">
  <div class="spinner" style="width: 20px; height: 20px;">
    <div class="spinner_bar" style="animation-delay: -1.2s; transform: rotate(0deg) translate(146%);"></div>
    <div class="spinner_bar" style="animation-delay: -1.1s; transform: rotate(30deg) translate(146%);"></div>
    <div class="spinner_bar" style="animation-delay: -1s; transform: rotate(60deg) translate(146%);"></div>
    <div class="spinner_bar" style="animation-delay: -0.9s; transform: rotate(90deg) translate(146%);"></div>
    <div class="spinner_bar" style="animation-delay: -0.8s; transform: rotate(120deg) translate(146%);"></div>
    <div class="spinner_bar" style="animation-delay: -0.7s; transform: rotate(150deg) translate(146%);"></div>
    <div class="spinner_bar" style="animation-delay: -0.6s; transform: rotate(180deg) translate(146%);"></div>
    <div class="spinner_bar" style="animation-delay: -0.5s; transform: rotate(210deg) translate(146%);"></div>
    <div class="spinner_bar" style="animation-delay: -0.4s; transform: rotate(240deg) translate(146%);"></div>
    <div class="spinner_bar" style="animation-delay: -0.3s; transform: rotate(270deg) translate(146%);"></div>
    <div class="spinner_bar" style="animation-delay: -0.2s; transform: rotate(300deg) translate(146%);"></div>
    <div class="spinner_bar" style="animation-delay: -0.1s; transform: rotate(330deg) translate(146%);"></div>
  </div>
</div>
`;
